### PR TITLE
Update webpacker

### LINKED ISSUE #43: Cannot run docker container w/ JiT Compilation in Docker For Mac
If running with the default `RAILS_ENV` (`development`), the process `bundle exec rails server` will throw a `EADDRNOTAVAIL`:

```
docker run -it --rm -p 3000:3000 my-image
=> Booting Puma
=> Rails 5.1.4 application starting in development
=> Run `rails server -h` for more startup options
Puma starting in single mode...
* Version 3.10.0 (ruby 2.4.2-p198), codename: Russell's Teapot
* Min threads: 5, max threads: 5
* Environment: development
* Listening on tcp://0.0.0.0:3000
Use Ctrl-C to stop
Started GET "/" for 172.17.0.1 at 2017-10-30 22:43:55 +0000
Cannot render console from 172.17.0.1! Allowed networks: 127.0.0.1, ::1, 127.0.0.0/127.255.255.255
Processing by HomeController#index as HTML
  Rendering home/index.html.erb within layouts/application
  Rendered home/index.html.erb within layouts/application (3.4ms)
Completed 500 Internal Server Error in 96ms



ActionView::Template::Error (Cannot assign requested address - connect(2) for [::1]:3035):
    1: <h2>Hello</h2>
    2: <%= javascript_pack_tag 'application' %>

app/views/home/index.html.erb:2:in `_app_views_home_index_html_erb___2910504907150917589_69962267300200'
2017-10-30 22:43:55 +0000: Rack app error handling request { GET / }
#<Errno::EADDRNOTAVAIL: Cannot assign requested address - connect(2) for [::1]:3035>
/usr/local/lib/ruby/2.4.0/socket.rb:1198:in `__connect_nonblock'
/usr/local/lib/ruby/2.4.0/socket.rb:1198:in `connect_nonblock'
/usr/local/lib/ruby/2.4.0/socket.rb:56:in `connect_internal'
/usr/local/lib/ruby/2.4.0/socket.rb:137:in `connect'
/usr/local/lib/ruby/2.4.0/socket.rb:627:in `block in tcp'
/usr/local/lib/ruby/2.4.0/socket.rb:227:in `each'
/usr/local/lib/ruby/2.4.0/socket.rb:227:in `foreach'
/usr/local/lib/ruby/2.4.0/socket.rb:617:in `tcp'
/usr/local/bundle/gems/webpacker-3.0.1/lib/webpacker/dev_server.rb:9:in `running?'
/usr/local/bundle/gems/webpacker-3.0.1/lib/webpacker/dev_server_proxy.rb:7:in `rewrite_response'
/usr/local/bundle/gems/rack-proxy-0.6.2/lib/rack/proxy.rb:57:in `call'
/usr/local/bundle/gems/railties-5.1.4/lib/rails/engine.rb:522:in `call'
/usr/local/bundle/gems/puma-3.10.0/lib/puma/configuration.rb:225:in `call'
/usr/local/bundle/gems/puma-3.10.0/lib/puma/server.rb:605:in `handle_request'
/usr/local/bundle/gems/puma-3.10.0/lib/puma/server.rb:437:in `process_client'
/usr/local/bundle/gems/puma-3.10.0/lib/puma/server.rb:301:in `block in run'
/usr/local/bundle/gems/puma-3.10.0/lib/puma/thread_pool.rb:120:in `block in spawn_thread'
2017-10-30 22:43:55 +0000: Rack app error handling request { GET /favicon.ico }
#<Errno::EADDRNOTAVAIL: Cannot assign requested address - connect(2) for [::1]:3035>
/usr/local/lib/ruby/2.4.0/socket.rb:1198:in `__connect_nonblock'
/usr/local/lib/ruby/2.4.0/socket.rb:1198:in `connect_nonblock'
/usr/local/lib/ruby/2.4.0/socket.rb:56:in `connect_internal'
/usr/local/lib/ruby/2.4.0/socket.rb:137:in `connect'
/usr/local/lib/ruby/2.4.0/socket.rb:627:in `block in tcp'
/usr/local/lib/ruby/2.4.0/socket.rb:227:in `each'
/usr/local/lib/ruby/2.4.0/socket.rb:227:in `foreach'
/usr/local/lib/ruby/2.4.0/socket.rb:617:in `tcp'
/usr/local/bundle/gems/webpacker-3.0.1/lib/webpacker/dev_server.rb:9:in `running?'
/usr/local/bundle/gems/webpacker-3.0.1/lib/webpacker/dev_server_proxy.rb:7:in `rewrite_response'
/usr/local/bundle/gems/rack-proxy-0.6.2/lib/rack/proxy.rb:57:in `call'
/usr/local/bundle/gems/railties-5.1.4/lib/rails/engine.rb:522:in `call'
/usr/local/bundle/gems/puma-3.10.0/lib/puma/configuration.rb:225:in `call'
/usr/local/bundle/gems/puma-3.10.0/lib/puma/server.rb:605:in `handle_request'
/usr/local/bundle/gems/puma-3.10.0/lib/puma/server.rb:437:in `process_client'
/usr/local/bundle/gems/puma-3.10.0/lib/puma/server.rb:301:in `block in run'
/usr/local/bundle/gems/puma-3.10.0/lib/puma/thread_pool.rb:120:in `block in spawn_thread'
^C- Gracefully stopping, waiting for requests to finish
=== puma shutdown: 2017-10-30 22:47:01 +0000 ===
- Goodbye!
Exiting
```

This has to do with `webpacker`'s Just in Time (JiT) webpack compilation. (See [Webpacker's docs](https://github.com/rails/webpacker#development)).

The `webpack-dev-server` and rails server collaborate (e.g.; HMR events) over websocket. We are using webpacker `3.0.1`. Currently, in order to change the bindings, command flags must be used. But it was not clear when running in vagrant/docker/etc how to property set this up.

With the [`webpacker v3.0.2`](https://github.com/rails/webpacker/blob/master/CHANGELOG.md#302---2017-10-04) we get some needed configuration hooks like *updated binstubs* and *dev server settings to be overriden by env variables*.

### PR DESCRIPTION
**Pull request by eckdanny-osi**
Resolves #43, Brings back added deps from #33, cleans up small docker nits

### Demonstration

The attached lice cap shows running a container built from master and from this branch:

fix-it

### CODE REVIEW

**Comment by edgibbs** [comment]
Can one of the admins verify this patch?

**Comment by eckdanny-osi** [comment]
This is kindofa bummer if we're wanting to bind-mount into the actual container, or do high fidelity testing locally. Updating to `3.0.1` made the `--arg` vs `WEBPACKER_DEV_SERVER_ARG` confusion totally moot.

File: package.json [hunk 1]
```diff
@@ -40,6 +42,7 @@
     "jest": "^21.2.1",
     "npm-run-all": "^4.1.1",
     "prettier": "^1.7.4",
-    "webpack-dev-server": "^2.9.1"
+    "webpack": "^3.0.0",
```
Comment (eckdanny-osi) [inline package.json]: from reverting #38

File: config/webpacker.yml [hunk 2]
```diff
@@ -33,11 +33,18 @@ development:
   <<: *default
   compile: true
 
+  # Reference: https://webpack.js.org/configuration/dev-server/
```
Comment (eckdanny-osi) [inline config/webpacker.yml]: 🤖 diffs in this file are from `bundle exec rails webpacker:install`

**Comment by eckdanny-osi** [comment]
_Oh!_ The reason why localhost or a native docker host (not Docker For Mac) works but publishing ports does not? 

 see https://github.com/rails/webpacker/issues/762
 see https://github.com/rails/webpacker/issues/767

**Pull request merged**